src/webgl/p5.RendererGL.js

```diff
@@ -1097,76 +1055,107 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
       this.filterShader = args[0];
     }
 
-    pg.clear(); // prevent undesirable feedback effects accumulating secretly
+    // Setting the target to the framebuffer when applying a filter to a framebuffer.
 
-    let pd = this._pInst.pixelDensity();
-    let texelSize = [1 / (this.width * pd), 1 / (this.height * pd)];
+    const target = this.activeFramebuffer() || this;
 
-    // apply blur shader with multiple passes
-    if (operation === constants.BLUR) {
+    // Resize the framebuffer 'fbo' and adjust its pixel density if it doesn't match the target.
+    this.matchSize(fbo, target);
+
+    // Set filterCamera for framebuffers.
+    if (target !== this) {
+      this.filterCamera = this.getFilterLayer().createCamera();
+    }
+    fbo.draw(() => this._pInst.clear()); // prevent undesirable feedback effects accumulating secretly.
+
+    let texelSize = [
+      1 / (target.width * target.pixelDensity()),
+      1 / (target.height * target.pixelDensity())
+    ];
 
-      const tmp = this.getFilterGraphicsLayerTemp();
-      tmp.clear(); // prevent feedback effects here too
+    // apply blur shader with multiple passes.
+    if (operation === constants.BLUR) {
+      // Treating 'tmp' as a framebuffer.
+      const tmp = this.getFilterLayerTemp();
+      // Resize the framebuffer 'tmp' and adjust its pixel density if it doesn't match the target.
+      this.matchSize(tmp, target);
+      tmp.draw(() => this._pInst.clear()); // prevent feedback effects here too
```

Comment:
Actually I think we don't need either this `clear` or the `image` call in line 1090 because we clear in the horizontal pass below. It's not breaking anything, but it's just doing extra work

Reply:
Yes, I was thinking that too. But in the previous code i found the same thing...I also thought to remove but as it was already written I did the same way for framebuffers. Okay gonna remove it.